libsentrykube/tests/test_quickpatch.py

```diff
@@ -0,0 +1,306 @@
+    ):
+        apply_patch(
+            _service,
+            _region,
+            _resource,
+            _patch,
+            {
+                "replicas1": num_replicas,
+            },
+        )
+    with pytest.raises(
+        ValidationError, match="Invalid arguments: 'replicas1' is a required property"
+    ):
+        apply_patch(
+            _service,
+            _region,
+            _resource,
+            _patch,
+            {
+                "replicas2": num_replicas,
+            },
+        )
+
+
+def test_missing_value_file():
+    with pytest.raises(
+        FileNotFoundError,
+        match=f"Resource value file not found for service {_service} in region {_region}",
+    ):
+        os.remove(
+            get_service_value_overrides_file_path(_service, _region, "default", False)
+        )
+        apply_patch(
+            _service,
+            _region,
+            _resource,
+            _patch,
+            {
+                "replicas1": num_replicas,
+                "replicas2": num_replicas,
+            },
+        )
+
+
+def test_invalid_resource():
+    with pytest.raises(
+        ValueError, match="Resource test-consumer-invalid is not allowed to be patched"
+    ):
+        apply_patch(
+            _service,
+            _region,
+            "test-consumer-invalid",
+            _patch,
+            {
+                "replicas1": num_replicas,
+                "replicas2": num_replicas,
+            },
+        )
+
+
+def test_correct_patch():
+    expected_data = """
+                    consumers:
+                        consumer:
+                            replicas: 10
+                    """
+    apply_patch(
+        _service,
+        _region,
+        _resource,
+        _patch,
+        {
+            "replicas1": num_replicas,
+            "replicas2": num_replicas,
+        },
+    )
+    config = get_service_value_overrides_file_path(_service, "us", "default", False)
+    actual = load_yaml(config)
+    expected = yaml.safe_load(expected_data)
+    assert expected == actual
+
+
+# TODO: Add more tests covering different patch operations and edge cases
+def test_missing_schema():
+    with pytest.raises(
+        ValueError,
+        match="Schema not found in patch file test-patch-missing-schema.yaml.j2",
+    ):
+        apply_patch(
+            _service,
+            _region,
+            _resource,
+            "test-patch-missing-schema",
+            {
+                "replicas1": num_replicas,
+                "replicas2": num_replicas,
+            },
+        )
+
+
+def test_invalid_arguments():
+    with pytest.raises(ValidationError):
+        apply_patch(
+            _service,
+            _region,
+            _resource,
+            _patch,
+            {
+                "replicas1": "invalid",  # Should be integer
+                "replicas2": num_replicas,
+            },
+        )
+
+
+def test_missing_required_argument():
```

Comment:
You can make all this more concise by using parametrize https://github.com/getsentry/sentry-infra-tools/blob/main/config_builder/test_materializer.py#L61-L83

As the test body is basically the same except for some parameters, you can define all the combinations of parameters and output in a list and pytest will run the same test on all